a8c-files.php

```diff
@@ -776,6 +776,232 @@ function a8c_files_init() {
+			list( $full_width, $full_height ) = wp_expand_dimensions( $image_width, $image_height, $full_width, $full_height );
+		}
+
+		$width_2x = round( $image_width * 2 );
+		$height_2x = round( $image_height * 2 );
+
+		// Add the current size and 2x current size. That should be the largest available size.
+		// The current "retina" browsers don't show images larger than 2x even on displays with 3.5x or 4.0x pixel density.
+		// Prepend wpcom-current. Fixes iOS8 bug.
+		$_sizes = array(
+			'vipgo-current' => array(
+				'width' => $image_width,
+				'height' => $image_height,
+			),
+			'vipgo-current-2x' => array(
+				'width' => $width_2x,
+				'height' => $height_2x,
+			),
+		) + $_sizes;
+
+		foreach ( $_sizes as $size_name => $size_data ) {
+			$already_added = false;
+			$_max_w = intval( $size_data['width'] );
+			$_max_h = intval( $size_data['height'] );
+
+			// Original image is too small.
+			if ( ! $_max_w || ( $full_width + 1 ) < $_max_w ) {
+				continue;
+			}
+
+			// Don't add default size larger than 2x the original image size (from the img tag).
+			if ( ( ! $_max_w || $_max_w > $width_2x ) && ( ! $_max_h || $_max_h > $height_2x ) ) {
+				continue;
+			}
+
+			list( $width, $height ) = wp_constrain_dimensions( $full_width, $full_height, $_max_w, $_max_h );
+
+			// Don't add images with +/- 2px size difference.
+			foreach ( $widths_added as $_w ) {
+				if ( abs( $width - $_w ) <= 2 ) {
+					$already_added = true;
+					break;
+				}
+			}
+
+			if ( $already_added ) {
+				continue;
+			}
+
+			$widths_added[] = $width;
+
+			$sizes[$size_name] = array(
+				'file' => self::set_photon_args( $photon_args, $width, $height, $image_basename ),
+				'width' => $width,
+				'height' => $height,
+				'mime-type' => $mime_type,
+			);
+		}
+
+		$image_meta['sizes'] = $sizes;
+		return $image_meta;
+	}
+
+	private static function get_thumbnail_sizes() {
+		$sizes = array(
+			'thumbnail' => array(
+				'width' => intval( get_option( 'thumbnail_size_w' ) ),
+				'height' => intval( get_option( 'thumbnail_size_h' ) ),
+			),
+			'medium' => array(
+				'width' => intval( get_option( 'medium_size_w' ) ),
+				'height' => intval( get_option( 'medium_size_h' ) ),
+			),
+			'medium_large' => array(
+					'width' => 768,
+					'height' => 0,
+				),
+				'large' => array(
+					'width' => intval( get_option( 'large_size_w' ) ),
+					'height' => intval( get_option( 'large_size_h' ) ),
+				),
+		);
+
+		if ( ! $sizes['thumbnail']['width'] && ! $sizes['thumbnail']['height'] ) {
+			$sizes['thumbnail']['width'] = 128;
+			$sizes['thumbnail']['height'] = 96;
+		}
+
+		if ( ! $sizes['medium']['width'] && ! $sizes['medium']['height'] ) {
+			$sizes['medium']['width'] = 300;
+			$sizes['medium']['height'] = 300;
+		}
+
+		if ( ! $sizes['large']['width'] && ! $sizes['large']['height'] ) {
+			$sizes['large']['width'] = 1024;
+			$sizes['large']['height'] = 0;
+		}
+
+		return apply_filters( 'vipgo_srcset_image_thumbnails', $sizes );
```

Comment:
Okay. Let's prefix this with `wpcom_vip_` instead of `vipgo_` anyway. We don't use `vipgo_` as a prefix on any filters currently, so consistency is a plus.